src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -194,15 +208,55 @@ private string GetResourceId()
             return this.GetPolicyArtifactFullyQualifiedId(this.Scope, Constants.MicrosoftAuthorizationPolicyAssignmentType, this.Name);
         }
 
+        /// <summary>
+        /// Verifies if identity type and/or identity ID and/or location are present
+        /// </summary>
+        private bool CheckIfIdentityPresent()
+        {
+            if (this.IdentityType != null & this.Location == null || this.AssignIdentity != null && this.AssignIdentity.IsPresent && this.Location == null)
+            {
+                throw new PSInvalidOperationException("Location needs to be specified if a managed identity is to be assigned to the policy assignment.");
+            }
+
+            if (this.IdentityType == ManagedIdentityType.UserAssigned && this.IdentityId == null)
```

Comment:
also use string.IsNullOrEmpty for identityId

Reply:
Fixed